runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/AbstractWindmillStream.java

```diff
@@ -167,14 +185,25 @@ private StreamObserver<RequestT> requestObserver() {
   }
 
   /** Send a request to the server. */
-  protected final void send(RequestT request) {
-    lastSendTimeMs.set(Instant.now().getMillis());
-    synchronized (this) {
-      if (streamClosed.get()) {
-        throw new IllegalStateException("Send called on a client closed stream.");
+  protected final synchronized void send(RequestT request) {
+    if (isShutdown()) {
+      return;
```

Comment:
should this throw instead of silently ignoring?

Reply:
we don't need to throw since the shutdown signal will propagate to the stream. This is for the case where `send()`s are queued up (we send this under a lock, so if there is an active send, subsequent calls to `send()` will be blocked until the active send returns).